testhelper/destination/kafka.go

```diff
@@ -119,7 +119,7 @@ func (k *KafkaResource) Destroy() error {
 	return g.Wait()
 }
 
-func SetupKafka(pool *dockertest.Pool, d deferer, opts ...Option) (*KafkaResource, error) {
+func SetupKafka(pool *dockertest.Pool, cln cleaner, opts ...Option) (*KafkaResource, error) {
```

Comment:
here we have a "Public" signature to require a private type `cleaner`, right?

Reply:
We don't require a private type, interfaces are just a contract. By not exporting the interface we're just saying "this is what we need but we offer no implementation so you'll have to choose one of your own". Meaning you have to use either the `*Cleanup` struct or `*testing.T` (or something that you can create yourself, as long as it has those two methods).